src/utils/filePathGenerator.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { join, sep } from 'path';
+import { MetadataComponent } from '../resolve/types';
+import { META_XML_SUFFIX } from '../common/constants';
+import { RegistryAccess } from '../registry/registryAccess';
+import { registry } from '..';
+const registryAccess = new RegistryAccess();
+
+export const filePathsFromMetadataComponent = (
```

Comment:
This is public, so needs js doc, particularly calling out the note you made in the PR description.